azure-pipelines.yml

```diff
@@ -209,12 +216,12 @@ jobs:
   - task: AzureArtifacts.manifest-generator-task.manifest-generator-task.ManifestGeneratorTask@0
     displayName: 'SBOM'
     inputs:
-      BuildDropPath: 'build_scripts/windows/out/'
+      BuildDropPath: 'build_scripts/windows/out/$(Platform)'
 
   - task: PublishPipelineArtifact@0
     displayName: 'Publish Artifact: MSI'
     inputs:
-      TargetPath: 'build_scripts/windows/out/'
+      TargetPath: 'build_scripts/windows/out/$(Platform)'
       ArtifactName: msi
```

Comment:
I am not entirely sure whether artifacts from different `matrix` (matrices) will be merged or overwritten. Let's give it a test.

Reply:
For posterity (since we talked internally about this briefly), the more changes I have to make to support building both x86 and x64 on a single machine, the more I'm leaning toward not having to support that and having separate artifacts would make more sense. It's all the little "gotchas" that worry me about changing the output paths radically, since that permiates throughout the build and pipelines.

Reply:
If `TargetPath` is `build_scripts/windows/out/$(Platform)`, x86 and x64 produce same file in that folder. Only one package will be saved.
I prefer jiasli's solution, it's more concise and less changes to made.

Reply:
> If `TargetPath` is `build_scripts/windows/out/$(Platform)`, x86 and x64 produce same file in that folder. Only one package will be saved.

Not really. In `build_scripts/windows/azure-cli.wixproj`,

```xml
<OutputPath>out\</OutputPath>
```

is changed to

```xml
<OutputPath>out\$(Platform)\</OutputPath>
```

So built MSIs are separated by `$(Platform)` folders.

Reply:
Keep things separate is quickly becoming a pervasive problem, though. A lot of scripts have to change, and workflows. I'm tempted to go back to a single output location and let separate agents build separate artifacts. I generally like to keep PRs focused on specific problems, and this feels a lot more like a separate effort to support multi-target builds across the build and publish pipeline.

Reply:
@jiasli I mean `TargetPath` needs to be set as `build_scripts/windows/out/` if we want to keep both builds in one artifact.